runtime/sys.c

```diff
@@ -588,16 +582,29 @@ CAMLprim value caml_sys_random_seed (value unit)
 #ifdef HAS_GETTIMEOFDAY
     struct timeval tv;
     gettimeofday(&tv, NULL);
-    data[n++] = tv.tv_usec;
-    data[n++] = tv.tv_sec;
+    if (n < 16) data[n++] = tv.tv_usec;
+    if (n < 16) data[n++] = tv.tv_sec;
 #else
-    data[n++] = time(NULL);
+    if (n < 16) data[n++] = time(NULL);
 #endif
 #ifdef HAS_UNISTD
-    data[n++] = getpid();
-    data[n++] = getppid();
+    if (n < 16) data[n++] = getpid();
+    if (n < 16) data[n++] = getppid();
```

Comment:
I can agree with the "check there is room before stuffing more data" logic.

I think the code would be even clearer like this:
```
  /* If the read from /dev/urandom fully succeeded, we now have 96 bits
     of good random data and can stop here. */
  if (n >= 12) return n;
  /* Otherwise, complement whatever we got (probably nothing)
     with some not-very-random data. */
#ifdef HAS_GETTIMEOFDAY
  struct timeval tv;
  gettimeofday(&tv, NULL);
   ...
```

Reply:
Yes, this was exactly my previous version, but this is not C99-msvc-whatever compliant due to the declaration after other statements.

Reply:
Blocks are your friends:
```
#ifdef HAS_GETTIMEOFDAY
  { struct timeval tv;
    gettimeofday(&tv, NULL);
    ... }
```

Reply:
"If you want" :-)